nova/scheduler/external.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2025 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+This module provides functionality to interact with an external scheduler API
+to reorder and filter hosts based on additional criteria.
+
+The external scheduler API is expected to take a list of weighed hosts and
+their weights, along with the request specification, and return a reordered
+and filtered list of host names.
+"""
+from oslo_log import log as logging
+import requests
+
+from nova.scheduler import utils
+
+LOG = logging.getLogger(__name__)
+
+
+def call_external_scheduler_api(weighed_hosts, weights, spec_obj, url):
+    """Reorder and filter hosts using an external scheduler service."""
+    if not url or not weighed_hosts:
+        return weighed_hosts
+    json_data = {
+        "spec": spec_obj.obj_to_primitive(),
+        "rebuild": utils.request_is_rebuild(spec_obj),
```

Comment:
Do we need to provide any of the other operations, too? We can also detect `live-migration` and `resize` at least. We also distinguish between baremetal and vmware VMs in scheduling - can the external scheduler handle getting called for Ironic nodes?

Reply:
Thanks for the hint, I'll add this to the request. I guess we'll start off picky in our external service and only run if specific conditions are met.